test/lib/utils.js

```diff
@@ -12,14 +12,20 @@ const Promise = require('bluebird');
 module.exports.withEnviron = function withEnviron(env, cb) {
   const origEnv = {};
   Object.keys(env).forEach((key) => {
-    origEnv[key] = process.env[key];
+    if (process.env.hasOwnProperty(key)) {
+      origEnv[key] = process.env[key];
+    }
     process.env[key] = env[key];
   });
   return Promise.resolve().then(() => {
     return cb();
   }).finally(() => {
     Object.keys(env).forEach((key) => {
-      process.env[key] = origEnv[key];
+      if (origEnv.hasOwnProperty(key)) {
+        process.env[key] = origEnv[key];
+      } else {
+        delete process.env[key];
+      }
```

Comment:
Previously, setting an env var that wasn't in `process.env`, would leave `process.env` with a property for that key with value `undefined`, which would leak into subsequent tests and cause convict to freak out when it couldn't convert `undefined` into the type for that env var.